### PR TITLE
Devel updates

### PR DESCRIPTION
**Pull request by marcomasa**
1. Firmware checkups for expected versions vs installed mbed versions at startup (ROS param usable)
2. Throttled debug output when safety stop is triggered
3. updated the output when init is not successful (deleted "check your code!" and replaced with useful suggestions)

### CODE REVIEW

**Comment by marcomasa** [comment]
updated branch:
- fixed issue #6 and added more missing deps
- added reset odometry service
- updated version and copyright date

**Comment by feesmrt** [comment]
Please test new code on the R&D robot @evocortex.

**Comment by marcomasa** [comment]
updated. everything is working now (tested with ohmnibot)

File: include/sw_interface/BaseControllerROS.h [hunk 1]
```diff
@@ -108,6 +113,10 @@ class BaseControllerROS
    void cbCmdVel(const geometry_msgs::Twist::ConstPtr& cmd_vel);
    void checkAndApplyCmdVel();
 
+   bool checkFirmwareVersion(const int major_ver, const int minor_ver, const int patch_ver);
```
Comment (feesmrt) [inline include/sw_interface/BaseControllerROS.h]: Firmware version is missleading. The evo-mbed library version is checked not the firmware version of the componentes. A firmware version check is already performend in the interface nodes of the devices.

File: src/sw_interface/BaseControllerROS.cpp [hunk 2]
```diff
@@ -440,20 +446,86 @@ void BaseControllerROS::publishLiftPos()
...
+                      << minor_ver << "] but evo_mbed is [" << EVO_MBED_TOOLS_VER_MINOR << "]!" <<  evo::error;
+      success &= false;
+   }
+
+   // patch is not critical, so dont end here
+   if(EVO_MBED_TOOLS_VER_PATCH < patch_ver)
+   {
+      evo::log::get() << _logger_prefix << "Detected Patch Version mismatch! Expected at least [" 
+                      << patch_ver << "] but evo_mbed is [" << EVO_MBED_TOOLS_VER_MINOR << "]!" <<  evo::warn;
+   }
```
Comment (feesmrt) [inline src/sw_interface/BaseControllerROS.cpp:504]: MAJOR Version: Compability to the previous version is not given. Check is ok.
MINOR Version: Backwards compability to previous version has to be ensured. So please chance the check to (EVO_MBED_TOOLS_VER_MINOR < minor_ver)

**Review by feesmrt — CHANGES_REQUESTED** [review]

File: src/sw_interface/BaseControllerROS.cpp (hunk 2 above)
Comment (marcomasa) [reply src/sw_interface/BaseControllerROS.cpp:504]: fixed this with commit   048a2e7

File: include/sw_interface/BaseControllerROS.h (hunk 1 above)
Comment (marcomasa) [reply include/sw_interface/BaseControllerROS.h]: fixed this with commit 048a2e7

File: src/sw_interface/BaseControllerROS.cpp (hunk 2 above)
Comment (feesmrt) [reply src/sw_interface/BaseControllerROS.cpp:504]: No value is returned from bool BaseControllerROS::checkMbedLibVersion()

"return success;" after patch level Test.

**Review by feesmrt — APPROVED** [review]

**Pull request merged**